features/steps/deciders.py

```diff
@@ -0,0 +1,64 @@
+class StepDecider:
+    """
+    Determines at the step level if a test should be run.
+    """
+
+    def __init__(self, context):
+        """
+        Behave context of currently running step.
+        :param context: A Behave framework context
+        """
+        self.context = context
+
+    def __skip_context(self):
+        return self.context.vendor_skip
```

Comment:
I'm still a bit concerned that for the 95% use case where we're *not* skipping, every step will be logging out this fact at https://github.com/sync-for-science/test-suite/blob/c346fe0427025e0580f84dfd349430beb67cb366/features/environment.py#L206 (with no logged information about which vendor or step, so the logs will just be filled with the same line over and over). Can we clean this up in this PR too?

Reply:
Also it's important to document how `ignored_steps` in our README and possibly in our example SMART config.